tests/libpod/sort.go

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package libpod
+
+import v1 "k8s.io/api/core/v1"
+
+// PodsByCreationTimestamp sorts a list of Pods by creation timestamp, using their names as a tie breaker.
+type PodsByCreationTimestamp []v1.Pod
```

Comment:
Looks good to me! But honestly I am wondering if it is worth it to make it more general that can be used elsewhere, with smth like:
```
import (
	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

// KMetaObject sorts a list of k8s meta objects by creation timestamp.
type KMetaObject []k8smetav1.ObjectMeta

func (o KMetaObject) Len() int      { return len(o) }
func (o KMetaObject) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
func (o KMetaObject) Less(i, j int) bool {
	if o[i].CreationTimestamp.Equal(&o[j].CreationTimestamp) {
		return o[i].Name < o[j].Name
	}
	return o[i].CreationTimestamp.Before(&o[j].CreationTimestamp)
}
```
I see that in the migration controller, we do basically the same thing, but for migrations https://github.com/kubevirt/kubevirt/blob/a6f4f91428c2acdc795ba7e9e8b20fa0d021244b/pkg/virt-controller/watch/migration.go#L1629-L1643
I don't know if it is worth it. WDYT?

Reply:
@fossedihelm Thanks for the feedback. The problem here is that the casting won't work: `sort.Reverse(PodsByCreationTimestamp(pods.Items)`

But we can solve it with Go generics. However IMO we would need some place like test-infra to put generic stuff there, and the libs would implement their things on top of that, it would require a separate PR though.

Reply:
That's fine! Thanks